### ISSUE TITLE
RFC: Inventory information fallback

### ISSUE THREAD

**Issue by harshalgohel** [lldpd/lldpd#498]
#### TL;DR
If `/sys/class/dmi/` does not exist (in case of Non-EFI bootloaders), then there could be some kind of fallback mechanism from which inventory information can be fetched.
I have mentioned some approaches for this, 

**What is community opinion on these?**

---

I can think of following approaches,

1. Fallback to linux environment variables
2. Configurable via `lldpcli` - hence configurable via `lldpd.conf`
3. Specify fallback directory 

---

### Fallback to environment variables,
1. Find inventory information from EEPROM/embedded flash
2. Write `/tmp/lldpd.env` or some other path.
3. Specify `EnvironmentFile=/tmp/lldpd.env` as part of systemd service. 
4. Export variables with `export $(cat /tmp/lldpd.env)` before executing lldpd for init.d and other init services.

### Each variable configurable via `lldpcli`
After `lldpd` is started,
1. Find inventory information from EEPROM/embedded flash
2. Execute `lldpcli configure inventory <inventory_var> <value>` for each of inventory variable such as `product_serial`, `product_name`, etc.

Before `lldpd` is started, 
1. Find inventory information from EEPROM/embedded flash
2. Write/append `configure inventory <inventory_var> <value>` to `lldpd.conf` 

### Fallback directory
1. Find inventory information from EEPROM/embedded flash
2. Write `/tmp/inventory/product_serial`, `/tmp/inventory/product_name`, etc.
3. Specify fallback directory with `lldpd.conf` or `lldpcli configure med fallback-directory /tmp/inventory/`
4. Specify fallback directory as compile time flag `./configure --fallback_dmi=/tmp/inventory`

**Comment by vincentbernat** [comment]
I would prefer only falling back to lldpcli and not adding yet another configuration mechanism.
